src/app/mat-main-nav/mat-main-nav.component.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { LayoutModule } from "@angular/cdk/layout";
+import { async, ComponentFixture, TestBed } from "@angular/core/testing";
+import { NoopAnimationsModule } from "@angular/platform-browser/animations";
+import { MatButtonModule } from "@angular/material/button";
+import { MatIconModule } from "@angular/material/icon";
+import { MatListModule } from "@angular/material/list";
+import { MatSidenavModule } from "@angular/material/sidenav";
+import { MatToolbarModule } from "@angular/material/toolbar";
+import { HttpClientModule } from "@angular/common/http";
+import { MatMainNavComponent } from "./mat-main-nav.component";
+
+describe("MatMainNavComponent", () => {
+  let component: MatMainNavComponent;
+  let fixture: ComponentFixture<MatMainNavComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [MatMainNavComponent],
+      imports: [
```

Comment:
you dont have to do any of these imports. you can simply add `schemas: [CUSTOM_ELEMENTS_SCHEMA]` to the test bed to get rid of the unknown element warnings

Reply:
done